src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs

```diff
@@ -272,16 +335,53 @@ public static IResourceBuilder<AzureEventHubsEmulatorResource> WithDataVolume(th
         => builder.WithVolume(name ?? VolumeNameGenerator.Generate(builder, "data"), "/data", isReadOnly: false);
 
     /// <summary>
-    /// Configures the gateway port for the Azure Event Hubs emulator.
+    /// Configures the host port for the Azure Event Hubs emulator is exposed on instead of using randomly assigned port.
     /// </summary>
     /// <param name="builder">Builder for the Azure Event Hubs emulator container</param>
-    /// <param name="port">Host port to bind to the emulator gateway port.</param>
+    /// <param name="port">The port to bind on the host. If <see langword="null"/> is used, a random port will be assigned.</param>
     /// <returns>Azure Event Hubs emulator resource builder.</returns>
+    [Obsolete("Use WithHostPort instead.")]
     public static IResourceBuilder<AzureEventHubsEmulatorResource> WithGatewayPort(this IResourceBuilder<AzureEventHubsEmulatorResource> builder, int? port)
+    {
+        return WithHostPort(builder, port);
+    }
+
+    /// <summary>
+    /// Configures the host port for the Azure Event Hubs emulator is exposed on instead of using randomly assigned port.
+    /// </summary>
+    /// <param name="builder">Builder for the Azure Event Hubs emulator container</param>
+    /// <param name="port">The port to bind on the host. If <see langword="null"/> is used, a random port will be assigned.</param>
+    /// <returns>Azure Event Hubs emulator resource builder.</returns>
+    public static IResourceBuilder<AzureEventHubsEmulatorResource> WithHostPort(this IResourceBuilder<AzureEventHubsEmulatorResource> builder, int? port)
     {
         return builder.WithEndpoint("emulator", endpoint =>
         {
             endpoint.Port = port;
         });
     }
+
+    /// <summary>
+    /// Adds a bind mount for the configuration file of an Azure Service Bus emulator resource.
+    /// </summary>
+    /// <param name="builder">The builder for the <see cref="AzureEventHubsEmulatorResource"/>.</param>
+    /// <param name="path">Path to the file on the AppHost where the emulator configuration is located.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureEventHubsEmulatorResource> WithConfigurationFile(this IResourceBuilder<AzureEventHubsEmulatorResource> builder, string path)
```

Comment:
Do we have a test for this method?

Reply:
It was also missing in SB, so I added it. At the same time I tried to simplify the implementation. However I intentionally didn't add support for a custom file provided for config, AND configuration hooks. I would assume if you provide a file, you could have updated it yourself. Hooks are useful when the emulator is generating a file from the resource model and you can't update some section from this model.